src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -304,47 +304,70 @@ def execute(self, args):
-                if cmd.supports_no_wait and getattr(expanded_arg, 'no_wait', False):
+                result = cmd_copy(params)
+                if cmd_copy.supports_no_wait and getattr(expanded_arg, 'no_wait', False):
                     result = None
-                elif cmd.no_wait_param and getattr(expanded_arg, cmd.no_wait_param, False):
+                elif cmd_copy.no_wait_param and getattr(expanded_arg, cmd_copy.no_wait_param, False):
                     result = None
 
-                transform_op = cmd.command_kwargs.get('transform', None)
+                transform_op = cmd_copy.command_kwargs.get('transform', None)
                 if transform_op:
                     result = transform_op(result)
 
                 if _is_poller(result):
-                    result = LongRunningOperation(self.cli_ctx, 'Starting {}'.format(cmd.name))(result)
+                    result = LongRunningOperation(cmd_copy.cli_ctx, 'Starting {}'.format(cmd_copy.name))(result)
                 elif _is_paged(result):
                     result = list(result)
 
                 result = todict(result, AzCliCommandInvoker.remove_additional_prop_layer)
                 event_data = {'result': result}
-                self.cli_ctx.raise_event(EVENT_INVOKER_TRANSFORM_RESULT, event_data=event_data)
-                result = event_data['result']
-                results.append(result)
-
+                cmd_copy.cli_ctx.raise_event(EVENT_INVOKER_TRANSFORM_RESULT, event_data=event_data)
+                return event_data['result']
             except Exception as ex:  # pylint: disable=broad-except
-                if cmd.exception_handler:
-                    cmd.exception_handler(ex)
+                if cmd_copy.exception_handler:
+                    cmd_copy.exception_handler(ex)
                     return CommandResultItem(None, exit_code=1, error=ex)
                 else:
                     six.reraise(*sys.exc_info())
 
+        from concurrent.futures import ThreadPoolExecutor, as_completed
+        tasks, results, exceptions = [], [], []
+        with ThreadPoolExecutor(max_workers=10) as executor:
+            for expanded_arg, cmd_copy in jobs:
+                tasks.append(executor.submit(_run_job, expanded_arg, cmd_copy))
+            for task in as_completed(tasks):
+                try:
+                    results.append(task.result())
+                except Exception as ex:  # pylint: disable=broad-except
+                    exceptions.append(ex)
+
+        # handle exceptions
+        if len(exceptions) == 1 and not results:
+            raise exceptions[0]
+        elif exceptions:
+            for exception in exceptions:
+                logger.warning(str(exception))
+            if not results:
+                return CommandResultItem(None, exit_code=1, error=CLIError('Encountered more than one exception.'))
+            else:
+                logger.warning('Encountered more than one exception.')
```

Comment:
This logic seems a little strange. Why do we use logger.warning when there's more than one exception instead of logger.error? Maybe a quick demo is warranted.